protocols/v2/binary-sv2/no-serde-sv2/codec/src/codec/decodable.rs

```diff
@@ -171,12 +227,20 @@ impl SizeHint for Vec<FieldMarker> {
     }
 }
 
+// Converts a `PrimitiveMarker` into a `FieldMarker`.
+//
+// Allows primitive types to be represented as field markers, which can
+// then be used in the decoding process.
```

Comment:
```suggestion
```

Converting from one type to another in very common in Rust and should be readable by devs as is